pkg/reconciler/contour/resources/httpproxy.go

```diff
@@ -171,17 +182,21 @@ func MakeHTTPProxies(ctx context.Context, ing *v1alpha1.Ingress, serviceToProtoc
 				} else {
 					postSplitHeaders = nil
 				}
-				var protocol *string
+
+				svc.RequestHeadersPolicy = postSplitHeaders
+
 				if proto, ok := serviceToProtocol[split.ServiceName]; ok {
-					protocol = ptr.String(proto)
+					svc.Protocol = ptr.String(proto)
 				}
-				svcs = append(svcs, v1.Service{
-					Name:                 split.ServiceName,
-					Port:                 split.ServicePort.IntValue(),
-					Weight:               int64(split.Percent),
-					RequestHeadersPolicy: postSplitHeaders,
-					Protocol:             protocol,
-				})
+
+				if cfg.Network != nil && cfg.Network.InternalEncryption {
+					svc.UpstreamValidation = &v1.UpstreamValidation{
+						CACertificate: fmt.Sprintf("%s/%s", "knative-serving", netcfg.ServingInternalCertName),
```

Comment:
Use - `system.Namespace()`

Since serving can be installed in other namespaces